### PR TITLE
add 6 system variables introduced in 6.3 but have no user doc

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/11378
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by crazycs520 — COMMENTED** [review]
LGTM

**Review by Defined2014 — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -909,6 +909,18 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Range: `[0, 9223372036854775807]`
 - This variable is used to set the number of retries when the DDL operation fails. When the number of retries exceeds the parameter value, the wrong DDL operation is canceled.
 
+### `tidb_ddl_flashback_concurrency` <span class="version-mark">New in v6.3.0</span>
+
+> **Warning:**
+>
+> The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.
```
Comment (qiancai) [inline system-variables.md:916]:
```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

Comment (qiancai) [reply system-variables.md:916]: "not recommended" or "do not",  please check it with R&D.

File: system-variables.md [hunk 2]
```diff
@@ -1814,6 +1866,18 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Default value: `100`
+- Range: `[1, 100000]`
+- This variable controls the maximum number of execution plans that can be cached using General Plan Cache.
```
Comment (qiancai) [inline system-variables.md:1879]:
```suggestion
- This variable controls the maximum number of execution plans that can be cached by General Plan Cache.
```

**Review by fzzf678 — APPROVED** [review]

**Comment by fzzf678** [comment]
LGTM

**Review by gengliqi — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -909,6 +909,18 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Range: `[0, 9223372036854775807]`
 - This variable is used to set the number of retries when the DDL operation fails. When the number of retries exceeds the parameter value, the wrong DDL operation is canceled.
```
[the remark this answers is not in the document]
Comment (shichun-0415) [reply system-variables.md:911]: Added in https://github.com/pingcap/docs/pull/9886.

File: system-variables.md [hunk 4]
```diff
@@ -1211,6 +1223,28 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - This variable is used to set whether to enable the statistics `Fast Analyze` feature.
 - If the statistics `Fast Analyze` feature is enabled, TiDB randomly samples about 10,000 rows of data as statistics. When the data is distributed unevenly or the data size is small, the statistics accuracy is low. This might lead to a non-optimal execution plan, for example, selecting a wrong index. If the execution time of the regular `Analyze` statement is acceptable, it is recommended to disable the `Fast Analyze` feature.
 
+### `tidb_enable_foreign_key` <span class="version-mark">New in v6.3.0</span>
+
+> **Warning:**
+>
+> The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.
```
Comment (shichun-0415) [inline system-variables.md:1230]:
```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

File: system-variables.md [hunk 5]
```diff
@@ -1211,6 +1223,28 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
...
+- Scope: GLOBAL
+- Persists to cluster: Yes
+- Default value: `OFF`
+- This variable controls whether to enable the `FOREIGN KEY` feature.
+
+### `tidb_enable_general_plan_cache` <span class="version-mark">New in v6.3.0</span>
+
+> **Warning:**
+>
+> The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.
```
Comment (shichun-0415) [inline system-variables.md:1241]:
```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

File: system-variables.md [hunk 6]
```diff
@@ -1496,6 +1537,17 @@ Query OK, 0 rows affected (0.09 sec)
 
 </CustomContent>
 
+### `tidb_enable_tiflash_read_for_write_stmt` <span class="version-mark">New in v6.3.0</span>
+
+> **Warning:**
+>
+> The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.
```
Comment (shichun-0415) [inline system-variables.md:1544]:
```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

File: system-variables.md [hunk 7]
```diff
@@ -1814,6 +1866,18 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 </CustomContent>
 
+### `tidb_general_plan_cache_size` <span class="version-mark">New in v6.3.0</span>
+
+> **Warning:**
+>
+> The feature controlled by this variable is not fully functional in the current version. It is **not recommended** that you change the default value.
```
Comment (shichun-0415) [inline system-variables.md:1873]:
```suggestion
> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
```

File: system-variables.md (hunk 1 above)
Comment (shichun-0415) [reply system-variables.md:916]: `Do not` is better.

File: system-variables.md [hunk 8]
```diff
@@ -1496,6 +1537,17 @@ Query OK, 0 rows affected (0.09 sec)
...
+### `tidb_enable_tiflash_read_for_write_stmt` <span class="version-mark">New in v6.3.0</span>
+
+> **Warning:**
+>
+> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Default value: `OFF`
+- This variable controls whether read requests in SQL write statements are pushed down to TiFlash.
```
Comment (qiancai) [inline system-variables.md:1549]:
```suggestion
- This variable controls whether read requests in SQL write statements can be pushed down to TiFlash.
```

**Review by qiancai — COMMENTED** [review]
LGTM

**Pull request merged**